ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/tabsonleft/v1/tabsonleft/clientlibs/site/js/tabs.js

```diff
@@ -0,0 +1,482 @@
+/*******************************************************************************
+ * Copyright 2023 Adobe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+(function () {
+
+    var keyCodes = {
+        END: 35,
+        HOME: 36,
+        ARROW_LEFT: 37,
+        ARROW_UP: 38,
+        ARROW_RIGHT: 39,
+        ARROW_DOWN: 40
+    };
+
+
+    class Tabs extends FormView.FormPanel {
```

Comment:
we should have a class `Tabs` in ui.frontend that extends Panel.
and then refactor Horizontal and Vertical tabs to extend from that class to avoid code duplication.